internal/provider/hypercore_vm_snapshot.go

```diff
@@ -0,0 +1,271 @@
+		MarkdownDescription: "Hypercore VM snapshot resource to manage VM snapshots",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "VM snapshot identifier",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"vm_uuid": schema.StringAttribute{
+				MarkdownDescription: "VM UUID of which we want to create a snapshot.",
+				Required:            true,
+			},
+			"type": schema.StringAttribute{
+				MarkdownDescription: "Snapshot type. Can be: USER, SUPPORT",
+				Optional:            true,
+				Computed:            true,
+			},
+			"label": schema.StringAttribute{
+				MarkdownDescription: "Snapshot label.",
+				Optional:            true,
+				Computed:            true,
+			},
+			"schedule_uuid": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "VM snapshot schedule identifier",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+func (r *HypercoreVMSnapshotResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	tflog.Info(ctx, "TTRT HypercoreVMSnapshotResource CONFIGURE")
+	// Prevent padisk if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	restClient, ok := req.ProviderData.(*utils.RestClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = restClient
+}
+
+func (r *HypercoreVMSnapshotResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Info(ctx, "TTRT HypercoreVMSnapshotResource CREATE")
+	var data HypercoreVMSnapshotResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if r.client == nil {
+		resp.Diagnostics.AddError(
+			"Unconfigured HTTP Client",
+			"Expected configured HTTP client. Please report this issue to the provider developers.",
+		)
+		return
+	}
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	restClient := *r.client
+	vmUUID := data.VmUUID.ValueString()
+	snapType := data.Type.ValueString()
+	snapLabel := data.Label.ValueString()
+
+	// Validate snapshot type
+	snapDiag := utils.ValidateSnapshotType(snapType)
+	if snapDiag != nil {
+		resp.Diagnostics.AddError(snapDiag.Summary(), snapDiag.Detail())
+		return
+	}
+
+	// Create VM snapshot
+	payload := map[string]any{
+		"domainUUID": vmUUID,
+		"label":      snapLabel,
+		"type":       snapType,
+
+		// These are all defaults from API
+		"automatedTriggerTimestamp":      0,
+		"localRetainUntilTimestamp":      0,
+		"remoteRetainUntilTimestamp":     0,
+		"blockCountDiffFromSerialNumber": -1,
```

Comment:
I think we should not POST those attributes. The `blockCountDiffFromSerialNumber` - it is for sure a read-only attribute, so do not POST it.

The `automatedTriggerTimestamp`, `localRetainUntilTimestamp`, `remoteRetainUntilTimestamp` - does web UI set those on POST?